cs-couchbase/src/main/java/io/cloudslang/content/couchbase/utils/InputsUtil.java

```diff
@@ -34,25 +48,47 @@ public static HttpClientInputs getHttpClientInputs(String username, String passw
         HttpClientInputs httpClientInputs = new HttpClientInputs();
-        httpClientInputs.setUsername(username);
-        httpClientInputs.setPassword(password);
-        httpClientInputs.setProxyHost(proxyHost);
-        httpClientInputs.setProxyPort(proxyPort);
-        httpClientInputs.setProxyUsername(proxyUsername);
-        httpClientInputs.setProxyPassword(proxyPassword);
-        httpClientInputs.setTrustAllRoots(trustAllRoots);
-        httpClientInputs.setX509HostnameVerifier(x509HostnameVerifier);
-        httpClientInputs.setTrustKeystore(trustKeystore);
-        httpClientInputs.setTrustPassword(trustPassword);
-        httpClientInputs.setKeystore(keystore);
-        httpClientInputs.setKeystorePassword(keystorePassword);
-        httpClientInputs.setConnectTimeout(connectTimeout);
-        httpClientInputs.setSocketTimeout(socketTimeout);
-        httpClientInputs.setUseCookies(useCookies);
-        httpClientInputs.setKeepAlive(keepAlive);
         httpClientInputs.setMethod(method);
         httpClientInputs.setAuthType(BASIC);
         httpClientInputs.setQueryParamsAreURLEncoded(Boolean.FALSE.toString());
+        httpClientInputs.setUsername(username);
+        httpClientInputs.setPassword(password);
+
+        httpClientInputs.setTrustAllRoots(valueOf(getEnforcedBooleanCondition(trustAllRoots, true)));
+        httpClientInputs.setKeepAlive(valueOf(getEnforcedBooleanCondition(keepAlive, true)));
+        httpClientInputs.setUseCookies(valueOf(getEnforcedBooleanCondition(useCookies, true)));
+
+        if (isNotBlank(proxyHost) && isNotBlank(proxyPort)) {
+            httpClientInputs.setProxyHost(proxyHost);
+            httpClientInputs.setProxyPort(proxyPort);
+        }
+
+        if (isNotBlank(proxyUsername) && isNotBlank(proxyPassword)) {
+            httpClientInputs.setProxyUsername(proxyUsername);
+            httpClientInputs.setProxyPassword(proxyPassword);
+        }
+
+        if (isNotBlank(trustKeystore) && isNotBlank(trustPassword)) {
+            httpClientInputs.setTrustKeystore(trustKeystore);
+            httpClientInputs.setTrustPassword(trustPassword);
+        }
+
+        if (isNotBlank(keystore) && isNotBlank(keystorePassword)) {
+            httpClientInputs.setKeystore(keystore);
+            httpClientInputs.setKeystorePassword(keystorePassword);
+        }
+
+        httpClientInputs.setConnectTimeout(getInputWithDefaultValue(connectTimeout, valueOf(INIT_INDEX)));
+        httpClientInputs.setSocketTimeout(getInputWithDefaultValue(socketTimeout, valueOf(INIT_INDEX)));
+
+        if (isBlank(x509HostnameVerifier)) {
+            httpClientInputs.setX509HostnameVerifier(ALLOW_ALL);
+        } else {
+            String[] hostnameVerifierValues = {ALLOW_ALL, BROWSER_COMPATIBLE, STRICT};
+            if (asList(hostnameVerifierValues).contains(x509HostnameVerifier)) {
```

Comment:
why did you use an array and converted it to a list instead of using `Arrays.asList(ALLOW_ALL, BROWSER_COMPATIBLE, STRICT).contains(x509HostnameVerifier)` (you'd avoid the variable assignment and the redundant conversion)
or if you still want to assign all in a variable, use 
`final List<String> hostnameVerifierValues = Arrays.asList(ALLOW_ALL, BROWSER_COMPATIBLE, STRICT)`
but i'd still go the first way